src/TSID/include/BipedalLocomotion/TSID/TaskSpaceInverseDynamics.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file TaskSpaceInverseDynamics.h
+ * @authors Ines Sorrentino
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_TASK_SPACE_INVERSE_DYNAMICS_H
+#define BIPEDAL_LOCOMOTION_TASK_SPACE_INVERSE_DYNAMICS_H
+
+#include <BipedalLocomotion/System/Advanceable.h>
+#include <BipedalLocomotion/System/LinearTask.h>
+
+#include <iDynTree/KinDynComputations.h>
+#include <manif/SE3.h>
+
+namespace BipedalLocomotion
+{
+
+namespace TSID
+{
+
+/**
+ * State of the TaskSpaceInverseDynamics
+ */
+ struct ContactWrench
+{
+    iDynTree::FrameIndex frameIndex; /**< Frame used to express the contact wrench */
+    System::VariablesHandler::VariableDescription variable; /**< Variable describing the contact
+                                                                wrench */
+};
+struct TSIDState
+{
+    manif::SE3d::Tangent baseAcceleration; /**< Mixed spatial acceleration of the base */
+    Eigen::VectorXd jointAccelerations; /**< Joints acceleration in rad per seconds */
+    Eigen::VectorXd jointTorques; /**< Joint torques */
+    std::vector<ContactWrench> contactWrenches; /**< List of the information related to the
+                                                     contact wrenches */
+};
+
+/**
+ * TaskSpaceInverseDynamics implements the interface for the task sapce inverse
+ * dynamics. Please inherits this class if you want to implement your custom Task TSID.
+ * The TaskSpaceInverseDynamics is a struct containing the joint acceleration, joint torques
+ * and contact wrenches. The TaskSpaceInverseDynamics can be used to generate the desired joint torques
+ * to be sent to the low-level torque controllers.
+ * Here you can find an example of the TaskSpaceInverseDynamics interface.
+ * <br/>
+ * <img src="https://user-images.githubusercontent.com/43743081/112606007-308f7780-8e18-11eb-875f-d8a7c4b960eb.png" width="1500">
+ */
+class TaskSpaceInverseDynamics : public BipedalLocomotion::System::Advanceable<TSIDState>
```

Comment:
A rebase with latest master will break this because of the recent changes introduced in https://github.com/dic-iit/bipedal-locomotion-framework/pull/267

Reply:
Ops I'm sorry! I fixed in acaa60c 620290b68ff5ca85d6520d55faad42d038d4aba6 and 620290b68ff5ca85d6520d55faad42d038d4aba6